ansible_ai_connect/ai/api/tests/test_views.py

```diff
@@ -2848,6 +2910,23 @@ def test_ok(self):
         self.assertEqual(r.data["format"], "plaintext")
         self.assertEqual(r.data["generationId"], generation_id)
 
+    @override_settings(ANSIBLE_AI_ENABLE_TECH_PREVIEW=True)
+    def test_ok_with_model_id(self):
+        generation_id = str(uuid.uuid4())
+        model = "mymodel"
+        payload = {
+            "text": "Install nginx on RHEL9",
+            "generationId": generation_id,
+            "ansibleExtensionVersion": "24.4.0",
+            "model": model,
+        }
+        self.client.force_authenticate(user=self.user)
+        r = self.client.post(reverse("generations"), payload, format="json")
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        self.assertIsNotNone(r.data["playbook"])
+        self.assertEqual(r.data["format"], "plaintext")
+        self.assertEqual(r.data["generationId"], generation_id)
```

Comment:
We're not asserting that `mymodel` was used.

Reply:
Will check using a segment event...